tests/backup/backup_rancher_test.go

```diff
@@ -1,6 +1,106 @@
 package tests
 
 import (
+	"fmt"
+	. "github.com/onsi/ginkgo"
+	"github.com/pborman/uuid"
+	"github.com/portworx/torpedo/drivers/backup"
+	"github.com/portworx/torpedo/drivers/scheduler"
+	"github.com/portworx/torpedo/drivers/scheduler/rke"
+	"github.com/portworx/torpedo/pkg/log"
+	. "github.com/portworx/torpedo/tests"
 	_ "github.com/rancher/norman/clientbase"
 	_ "github.com/rancher/rancher/pkg/client/generated/management/v3"
+	"os"
+	"strings"
+	"time"
 )
+
+// This testcase creates basic rancher project creation and adding ns to that project
+var _ = Describe("{BasicRancherTest}", func() {
+
+	var (
+		appNamespaces []string
+		contexts      []*scheduler.Context
+		appContexts   []*scheduler.Context
+	)
+
+	JustBeforeEach(func() {
+		StartTorpedoTest("BasicRancherTestcase",
+			"Creates basic rancher project creation and adding ns to that project", nil, 0)
+		log.InfoD("Deploying applications required for the testcase")
+		contexts = make([]*scheduler.Context, 0)
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			taskName := fmt.Sprintf("%s-%d", taskNamePrefix, i)
+			appContexts = ScheduleApplications(taskName)
+			contexts = append(contexts, appContexts...)
+			for _, ctx := range appContexts {
+				ctx.ReadinessTimeout = appReadinessTimeout
+				namespace := GetAppNamespace(ctx, taskName)
+				appNamespaces = append(appNamespaces, namespace)
+			}
+		}
+	})
+
+	It("Basic Backup Creation", func() {
+		ctx, err := backup.GetAdminCtxFromSecret()
+		log.FailOnError(err, "Fetching px-central-admin ctx")
+		Step("Creating backup location and cloud setting", func() {
+			log.InfoD("Creating backup location and cloud setting")
+			backupLocationProviders := getProviders()
+			kubeconfigs := os.Getenv("KUBECONFIGS")
+			kubeconfigList := strings.Split(kubeconfigs, ",")
+			for _, kubeconfig := range kubeconfigList {
+				for _, provider := range backupLocationProviders {
+					credName := fmt.Sprintf("autogenerated-cred-%v", time.Now().Unix())
+					//cc := AssignClusterCloudCred("", provider)
+					credUid := uuid.New()
+					err := CreateCloudCredentialNew(provider, ctx, orgID, credName, credUid, kubeconfig)
+					dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying creation of cloud credential named [%s] for org [%s]  as provider for backup location", credName, orgID))
+					customBackupLocationName := fmt.Sprintf("autogenerated-backup-location-%v", time.Now().Unix())
+					backupLocationUID := uuid.New()
+					err = CreateBackupLocation(provider, customBackupLocationName, backupLocationUID, credName, credUid, getGlobalBucketName(provider), orgID, "")
+					dash.VerifyFatal(err, nil, fmt.Sprintf("Creating backup location %s", customBackupLocationName))
+				}
+			}
+		})
```

Comment:
What if we incorporate this logic inside the CreateCloudCredential function?

Reply:
Yes Yes. Since I did not want other testcases to be affected now, have introduced a new method.
Eventually there will be only CreateCloudCredential for all the testcases